plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsMaker.java

```diff
@@ -23,38 +27,57 @@
 import io.trino.spi.type.TypeManager;
 import org.apache.iceberg.FileScanTask;
 import org.apache.iceberg.Schema;
+import org.apache.iceberg.StatisticsFile;
 import org.apache.iceberg.Table;
 import org.apache.iceberg.TableScan;
 import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.stats.BlobMetadata;
+import org.apache.iceberg.stats.FileMetadata;
+import org.apache.iceberg.stats.StatsReader;
+import org.apache.iceberg.types.Type;
 import org.apache.iceberg.types.Types;
 
+import javax.inject.Inject;
+
 import java.io.IOException;
 import java.io.UncheckedIOException;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static com.google.common.collect.Streams.stream;
 import static io.trino.plugin.iceberg.ExpressionConverter.toIcebergExpression;
 import static io.trino.plugin.iceberg.IcebergUtil.getColumns;
+import static io.trino.plugin.iceberg.IcebergUtil.primitiveFieldTypes;
+import static io.trino.plugin.iceberg.Stats.NDV_STATS;
+import static io.trino.plugin.iceberg.Stats.bytesLittleEndianToLong;
+import static io.trino.plugin.iceberg.TypeConverter.toTrinoType;
+import static java.util.Comparator.comparing;
+import static java.util.Objects.requireNonNull;
 import static java.util.function.Function.identity;
 import static java.util.stream.Collectors.toUnmodifiableMap;
 
 public class TableStatisticsMaker
```

Comment:
Is it still maker? It looks much more like Reader

Reply:
it's maker based off of what's available in table metadata.
i am fine with "reader" as well